core-bundle/src/Migration/Version500/AccesskeyMigration.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version500;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Types\BooleanType;
+
+/**
+ * Removes '0' from tl_page.accesskey and tl_form_field.accesskey as this was accidentally introduced (see #5586).
+ */
+class AccesskeyMigration extends AbstractMigration
+{
+    private static array $affectedTables = ['tl_page', 'tl_form_field'];
+
+    public function __construct(private Connection $connection)
+    {
+    }
+
+    public function shouldRun(): bool
+    {
+        foreach (self::$affectedTables as $table) {
+            if ($this->needsMigration($table)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function run(): MigrationResult
+    {
+        foreach (self::$affectedTables as $table) {
+            if ($this->needsMigration($table)) {
+                $this->connection->executeQuery("ALTER TABLE $table CHANGE accesskey accesskey CHAR(1) DEFAULT '' NOT NULL");
+                $this->connection->executeQuery("UPDATE $table SET accesskey = '' WHERE accesskey = '0'");
```

Comment:
This would reset my access key `0` that I am using in my navigation menu.

Reply:
In Contao 5.0, until this fix, all access keys are broken anyways.

If updating from an older database this migration would not be executed, because it only runs if the column type is boolean.

Reply:
I would prefer not to add the migration. Only 1% of the users are already using Contao 5 and potentially have the issue, but the migration will break the access key `0` for the remaining 99% once they upgrade to Contao 5.

Reply:
> but the migration will break the access key `0` for the remaining 99% once they upgrade to Contao 5.

No, this migration will never be executed for them.

> Only 1% of the users are already using Contao 5 and potentially have the issue

All Contao 5 users have an accesskey of `0` for every single page, they would all have to fix this manually anyways.